lxd/migration/migration_volumes.go

```diff
@@ -0,0 +1,181 @@
+package migration
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/ioprogress"
+	"github.com/lxc/lxd/shared/units"
+)
+
+// Type represents the migration transport type. It indicates the method by which the migration can
+// take place and what optional features are available.
+type Type struct {
+	FSType   MigrationFSType
+	Features []string
+}
+
+// ToHeader converts a Type to a MigrationHeader.
+func (t *Type) ToHeader() MigrationHeader {
```

Comment:
One thing to be careful with here is that we'll want the rsync features to always be set.

That's needed so that if the target decides that it can't use the fstype of the source, it can compare the rsync features of the source with its own, find a matching set, override the features list and then return the header struct.

If the source only gives us the zfs features (for example), then we can't negotiate the rsync features in the case of a fs mismatch.

Reply:
@stgraber I have reworked this now. 

Because the `Type` struct (by design) can not differentiate between rsync features and zfs features in the same struct, then I've modified the header generation and match functions to use the header struct directly.

I'e made the `TypesToHeader` variadic so it can be used both with generating an offer header using multiple Types from the pool in preference order, as well as sending back a negotiated header generated from the singular type that the target decided upon.